### PR TITLE
[WIP] chroot-rootfs: Add section.

### PR DESCRIPTION
**Pull request by flexibeast**
Further to @Duncaen's [comment](https://github.com/void-linux/void-docs/issues/267#issuecomment-626345328) on #267.

### CODE REVIEW

**Comment by ericonr** [comment]
We do have xtool's `xchroot`, which does basically those steps. How do you feel about recommending it?

**Comment by ericonr** [comment]
I'd mention you should create and populate the directory as root, otherwise permissions tend to get weird. And I'm not entirely sure how we could fit `xbps-uunshare` as a non-privileged tool, or even if we need to.

**Comment by st3r4g** [comment]
In order to run graphical applications (imagine a steam chroot on a host musl system) one must also mount `/tmp` (for X11 and Xwayland) and/or `/run` (for Wayland)

**Comment by Piraty** [comment]
how about extending it to be more generic regarding target architecture?
building a bootable image (either same arch or foreign arch, like rpi ) involves some of the same steps but you install `base-system` (or `rpi-base`) instead of `base-voidstrap`

File: src/installation/musl.md [hunk 1]
```diff
@@ -25,28 +25,5 @@ which provide their own libc in the image.
...
-then run glibc programs as usual. Once you've finished using it, unmount the
-chroot using [umount(8)](https://man.voidlinux.org/umount.8).
-
-#### PRoot
-
-An alternative to the above is [proot(1)](https://man.voidlinux.org/proot.1), a
-user-space implementation of chroot, mount --bind, and binfmt_misc. By
-installing the `proot` package, unprivileged users can utilize a chroot
-environment.
+Software requiring glibc can be run in a glibc chroot. For further information,
```
Comment (Piraty) [inline src/installation/musl.md:28]: just put the reference under `chroot` keyword

**Comment by jeffayle** [comment]
> but you install base-system (or rpi-base) instead of base-voidstrap

I don't think this is true if you're not building a bootable system. `rpi-base` includes a kernel, which you definitely don't need in this situation. There might be something else in there that you need to get it working though?

**Comment by flexibeast** [comment]
Just to be clear, @Duncaen's comment that led me to create this section was:

> I was thinking more about a "How to setup Void Linux container/chroot rootfs", which I think is useful and quite different from installing void linux on bare metal using a chroot.

There's a good chance i've misinterpreted @Duncaen's comment, but i assumed what was meant was a page about setting up a chroot on an existing Void installation (e.g. a glibc chroot on a musl system, currently in the Handbook [here](https://docs.voidlinux.org/config/musl.html)). Installations via a chroot will be covered by #177 (which is almost complete, and which i feel will be a crucial addition to the Handbook), and then by further work leveraging #177 to address #85.

@ericonr: i wasn't aware of `xchroot` as part of `xtools`, as even though `man xchroot` takes me to the `xtools` man page, there seems to be no mention of it? (And the layout of the `xtools` man page [on man.voidlinux.org](https://man.voidlinux.org/xtools.1) is impressively broken.) Given what i've written above, is it a tool that can be used in this context? If so, the `xtools` man page will need to be updated, and of course, the `xmandump` stuff will need to be resolved so that we can point to the updated man page.

File: src/installation/musl.md (hunk 1 above)
Comment (flexibeast) [reply src/installation/musl.md:28]: Sure, will do.

**Comment by jeffayle** [comment]
To what extent should the handbook document *what* to do with a chroot? glibc compatibility on musl definitely makes sense (apart from already being in the handbook) because very few distributions support musl. Other uses might include running x86 binaries on x86_64 distribution (maybe this should go in the multilib page?) and creating jails for non-trusted users (if this is still good advice with docker and lxc around?)

**Comment by ericonr** [comment]
@flexibeast master of xtools has a reference to `xchroot`. It should be usable in that context, but perhaps we can mention it as a "tool that automates this process", and still document the process. I have updated `xmandump`, and maldrige is working on getting it up and running c:

@st3r4g you'd need to bind-mount /tmp for X11 to work, right? In that case, `xchroot` is definitely not the tool.

**Comment by st3r4g** [comment]
the wiki said to install `xhost` and run `xhost +local:`, it worked, but I don't know all the implications of that. Bind-mounting `/tmp` (or maybe just the X socket inside tmp) seems a better suggestion, do you agree?

**Comment by ericonr** [comment]
@st3r4g if `xhost` "just works", it might be a better solution. Did you test how well bind-mounting `/tmp` worked? It *is* a more complicated step and can be easy to get wrong, I think.

**Comment by st3r4g** [comment]
Yes, I'm currently using a script that binds mount `/tmp` too, that way I also get `/tmp` to be a `tmpfs` inside the chroot. I don't think is more complicated really

**Comment by ericonr** [comment]
I'm not sure what I'd prefer to see here, and also don't know what's safest/more correct. Thanks for testing :p

**Comment by flexibeast** [comment]
Referring to `xvoidstrap` and `xchroot` does seem to be the way to go, although there seem to be a few blockers involved:

* `xchroot` does `mount --rbind /proc "$CHROOT/proc"; mount --rbind /sys "$CHROOT/sys"` rather than the `mount` commands described by @q66 in [this comment](https://github.com/void-linux/void-docs/pull/163#discussion_r395958797). i don't know if the source for `xchroot` needs to be changed, or whether instead it's fine for the contexts in which it's intended to be used. (What are those contexts?)

* At any rate, there needs to be a release of `xtools` where the man page references `xchroot` (which it does on `xtools` [master](https://github.com/leahneukirchen/xtools)).

* The layout of [the xtools.1 man page on man.voidlinux.org](https://man.voidlinux.org/xtools.1) needs to be fixed.

**Comment by ericonr** [comment]
- latest version with `xchroot` in the man page was released a few days ago
- I opened an issue about the layout here https://github.com/leahneukirchen/xtools/issues/184, but we don't know what's causing it

**Comment by flexibeast** [comment]
@ericonr: Ah, thanks, i hadn't noticed the `xtools` package update.

**Comment by flexibeast** [comment]
To help keep track of what needs to be done in relation to this section, a list:

[ ] Mention need to mount `/tmp`, unless it should be done via `xchroot` (cf. [this comment](https://github.com/void-linux/void-docs/pull/300#issuecomment-632864663))
[ ] Ensure section is applicable to multiple architectures (cf. [this comment](https://github.com/void-linux/void-docs/pull/300#issuecomment-632887934) and [this comment](https://github.com/void-linux/void-docs/pull/300#issuecomment-632937264))
[ ] Confirm whether the source `xchroot` needs to be modified re. `mount` commands (cf. [this comment](https://github.com/void-linux/void-docs/pull/300#issuecomment-636569842))
[ ] Confirm appropriate use of `xhost` (cf. [this comment](https://github.com/void-linux/void-docs/pull/300#issuecomment-634462437))
[ ] Fix layout of `xtools` man page on man.voidlinux.org (cf. [this issue](https://github.com/leahneukirchen/xtools/issues/184))

**Comment by ericonr** [comment]
I think we should leave `/tmp` as a manual mount. `xchroot` doesn't need to always do it, and a user can choose to either have it as a separate tmpfs or as a bind mount. For this use case, it should be a bind mount.

I believe installing base-system is enough. I have a glibc chroot on my musl RPi. If you need graphics support or something, then you can install `rpi-base` - even then, I don't believe it's needed.

**Comment by flexibeast** [comment]
Blocked on #413.

**Comment by flexibeast** [comment]
@ericonr: Would you be interested in taking over this PR?

**Comment by ericonr** [comment]
I can try!

**Comment by flexibeast** [comment]
Thanks!

File: src/config/chroot-rootfs.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Creating a chroot rootfs
+
+As root, create a directory that will contain the chroot, and install a base
```
Comment (Piraty) [inline src/config/chroot-rootfs.md:3]: it is not necessary that the chroot dir is owned by root.
```
$ XBPS_ARCH=x86_64-musl xbps-install -MS --rootdir=/tmp/rootfs.void.x86_64-musl --repository=https://alpha.de.repo.voidlinux.org/current/musl base-voidstrap
$ proot -R /tmp/rootfs.void.x86_64-musl/
```

**Pull request closed without merging**